plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1608,17 +1619,31 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
       try {
          if( !remove_expired_trxs( preprocess_deadline ) )
             return start_block_result::exhausted;
+         
+         if (!complete_produced_block_if_ready())
+            return start_block_result::failed;
+         
          if( !remove_expired_blacklisted_trxs( preprocess_deadline ) )
             return start_block_result::exhausted;
+         
+         if (!complete_produced_block_if_ready())
+            return start_block_result::failed;
+         
          if( !_subjective_billing.remove_expired( _log, chain.pending_block_time(), fc::time_point::now(), preprocess_deadline ) )
             return start_block_result::exhausted;
 
+         if (!complete_produced_block_if_ready())
+            return start_block_result::failed;
+
          // limit execution of pending incoming to once per block
          size_t pending_incoming_process_limit = _unapplied_transactions.incoming_size();
 
          if( !process_unapplied_trxs( preprocess_deadline ) )
```

Comment:
Need to check `complete_produced_block_if_ready` within the loop of `process_unapplied_trxs`.